src/Analysis/Ast/Impl/Analyzer/Definitions/IPythonAnalyzer.cs

```diff
@@ -20,14 +20,15 @@
 using Microsoft.Python.Analysis.Diagnostics;
 using Microsoft.Python.Analysis.Types;
 using Microsoft.Python.Core.Collections;
+using Microsoft.Python.Parsing.Ast;
 
 namespace Microsoft.Python.Analysis.Analyzer {
     public interface IPythonAnalyzer {
         Task WaitForCompleteAnalysisAsync(CancellationToken cancellationToken = default);
         /// <summary>
         /// Schedules module for analysis. Module will be scheduled if version of AST is greater than the one used to get previous analysis
         /// </summary>
-        void EnqueueDocumentForAnalysis(IPythonModule module, int version);
+        void EnqueueDocumentForAnalysis(IPythonModule module, PythonAst ast, int version);
```

Comment:
There were several cases when `ast` was out of sync, so we need to pass to `PythonAnalyzer` explicitly.

Reply:
Wouldn't this indicate some other race condition, if the module's Ast is changing during an analysis? When would this occur? When the user modifies their code?

Reply:
Any change to the file, including opening/closing it. I'm not sure that having separate Parse/Analyze queue is still valuable for us, cause we don't have analysis on demand anymore - we use a graph, but that would be a separate change that I don't want to make until database support is in master.